### PR TITLE
Implement search create #487

### PR DESCRIPTION
**Pull request by mkshah605**

### CODE REVIEW

File: tests/data/search_result_20200130_093253_ssc6_u0001.json [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+        "strip_id": "s106_20200130T093253Z",
+        "sun_azimuth": 202.2,
+        "sun_elevation": 42.6,
+        "updated": "2020-01-31T13:40:08Z",
+        "view_angle": 15.8,
+        "visible_confidence_percent": 89,
+        "visible_percent": 100
+    },
+    "type": "Feature"
+}
```
Comment (jreiberkyle) [inline tests/data/search_result_20200130_093253_ssc6_u0001.json:497]: May want to make the editors happy by adding the newline here

**Review by jreiberkyle — CHANGES_REQUESTED** [review]
looks great! A few suggestions but the core logic is spot on.

File: tests/integration/test_data_cli.py [hunk 2]
```diff
@@ -79,6 +81,95 @@ def test_data_search_quick_filter_success(invoke, item_types):
...
+            "config": {
+                "gt": "2019-12-31T00:00:00Z", "lte": "2020-01-31T00:00:00Z"
+            }
+        },
+        "item_types": ["SkySatScene"],
+        "__daily_email_enabled": True
+    }
+    sent_request = json.loads(respx.calls.last.request.content)
+    assert not result.exception
+    assert sent_request == order_request
```
[the remark this answers is not in the document]
Comment (mkshah605) [reply tests/integration/test_data_cli.py:170]: Thanks Jen these are great suggestions! By unedited, do you mean not pretty printed? Do you have recommendations for how to check this besides just checking for a specific string?

File: tests/data/search_result_20200130_093253_ssc6_u0001.json (hunk 1 above)
Comment (mkshah605) [reply tests/data/search_result_20200130_093253_ssc6_u0001.json:497]: where should the newline go?

Comment (jreiberkyle) [reply tests/data/search_result_20200130_093253_ssc6_u0001.json:497]: At the end of the doc. It doesn't show it in the commends, but if you look at the diff you will see the red circle and the hover message indicates the issue is no newline.

image

File: tests/integration/test_data_cli.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jreiberkyle) [reply tests/integration/test_data_cli.py:170]: Something like this: https://github.com/planetlabs/planet-client-python/blob/c90c06dfdaf4ff3b4f7a5ce97e63a628fb16b442/tests/integration/test_orders_cli.py#L108

File: tests/data/search_result_20200130_093253_ssc6_u0001.json (hunk 1 above)
Comment (mkshah605) [reply tests/data/search_result_20200130_093253_ssc6_u0001.json]: code updated!

File: tests/integration/test_data_cli.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkshah605) [reply tests/integration/test_data_cli.py]: code updated!

File: planet/cli/data.py [hunk 3]
```diff
@@ -86,18 +86,14 @@ def parse_filter(ctx, param, value: str) -> dict:
...
-    Arguments: \n
-    ITEM_TYPES - string. Comma-separated item type identifier(s). \n
-    FILTER - string. A full JSON description of search criteria.
-    Supports file and stdin. \n
-
-    Output:
-    A series of GeoJSON descriptions for each of the returned items.
-
+    parameter will be applied to the stored quick search. This function
+    outputs a series of GeoJSON descriptions for each of the returned items.
```
Comment (jreiberkyle) [inline planet/cli/data.py:93]:
```suggestion
    outputs a series of GeoJSON descriptions, one for each of the returned items.
```

File: planet/cli/data.py [hunk 4]
```diff
@@ -86,18 +86,14 @@ def parse_filter(ctx, param, value: str) -> dict:
...
+    The number of output items can be limited usig the "--limit" option,
+    which defaults to 100. The output can also be optionally pretty-printed
```
Comment (jreiberkyle) [inline planet/cli/data.py:95]:
```suggestion
    The limit on the number of output items can be controlled using the "--limit" option, which defaults to 100. If "--limit" is set to zero, no limit is applied and all results (a potentially large number) are returned. The output can also be optionally pretty-printed
```

File: tests/conftest.py [hunk 5]
```diff
@@ -54,6 +54,12 @@ def func(filename):
     return func
 
 
+@pytest.fixture
+def search_result(get_test_file_json):
```
Comment (jreiberkyle) [inline tests/conftest.py:58]: really nice re-use of fixtures here.

File: tests/integration/test_data_cli.py [hunk 6]
```diff
@@ -79,6 +81,96 @@ def test_data_search_quick_filter_success(invoke, item_types):
...
+                                   "key": "value"
+                               }]})
+    respx.post(TEST_SEARCHES_URL).return_value = mock_resp
+
+    runner = CliRunner()
+    result = invoke(["search-create", name, item_types, json.dumps(filter)],
+                    runner=runner)
+
+    assert result.exit_code == 0
+    assert len(result.output.strip().split('\n')) == 1  # we have 1 feature
```
Comment (jreiberkyle) [inline tests/integration/test_data_cli.py:132]: thank you for this helpful comment here. well placed :)

File: tests/integration/test_data_cli.py [hunk 7]
```diff
@@ -79,6 +81,96 @@ def test_data_search_quick_filter_success(invoke, item_types):
...
+                "gt": "2019-12-31T00:00:00Z", "lte": "2020-01-31T00:00:00Z"
+            }
+        },
+        "item_types": ["SkySatScene"],
+        "__daily_email_enabled": True
+    }
+    sent_request = json.loads(respx.calls.last.request.content)
+    assert result.exit_code == 0
+    assert sent_request == search_request
+    assert json.loads(result.output) == search_result
```
Comment (jreiberkyle) [inline tests/integration/test_data_cli.py:171]: excellent set of assertions. complete while only testing what is important. nicely done.

**Review by jreiberkyle — APPROVED** [review]
looks SO GOOD! nicely done! I suggested a few tweaks to the docs wording. LGTM.

**Pull request merged**